hack/test-full.sh

```diff
@@ -0,0 +1,255 @@
+#
+# Also note that when run with all default options, the "Matrix"
+# test will run, requiring that all supported language toolchains are
+# also available.
+#
+# For more targeted E2E testing without all the bells-and-whistles,
+# see the e2e/e2e_test.go file which can have it's tests run directly.
+#
+# make test-full 2>&1 | tee ./test-full.log
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source "$(cd "$(dirname "$0")" && pwd)/common.sh"
+
+# Enable Optional Tests
+# ---------------------
+# The defaults in the e2e test implementation are a bit more conservative.
+# Here we toggle on All The Things.  Note that we still allow any settings
+# made explicitly in the current environment to take precidence; just setting
+# new defaults which are more expansive in testing scope.
+export FUNC_CLUSTER_RETRIES="${FUNC_CLUSTER_RETRIES:-5}"
+export FUNC_E2E_MATRIX="${FUNC_E2E_MATRIX:-true}"
+export FUNC_E2E_VERBOSE="${FUNC_E2E_VERBOSE:-true}"
+export FUNC_E2E_PODMAN="${FUNC_E2E_PODMAN:-true}"
+export FUNC_INT_TEKTON_ENABLED="${FUNC_INT_TEKTON_ENABLED:-1}"
+export FUNC_INT_GITLAB_ENABLED="${FUNC_INT_GITLAB_ENABLED:-1}"
+export FUNC_INT_GITLAB_HOSTNAME="${FUNC_INT_GITLAB_HOSTNAME:-gitlab.localtest.me}"
+export FUNC_INT_PAC_HOST="${FUNC_INT_PAC_HOST:-pac-ctr.localtest.me}"
+
+main() {
+    setup         # paths etc
+    preconditions # binaries exist, cluster available etc
+
+    # 1:1 for GitHub Workflow Jobs of the same name:
+    precheck
+    test-unit
+    test-templates
+    test-integration
+    test-e2e
+    test-e2e-podman
+    test-e2e-runtimes
+}
+
+# -----
+# Setup
+# -----
+setup() {
+    # Determine script directory
+    SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+    PROJECT_ROOT="$(cd "${SCRIPT_DIR}/.." && pwd)"
+
+    # Set PATH and KUBECONFIG
+    export PATH="${PROJECT_ROOT}/hack/bin:${PATH}"
+    export KUBECONFIG="${KUBECONFIG:-${PROJECT_ROOT}/hack/bin/kubeconfig.yaml}"
+
+    # GitLab test configuration
+    # This is the default set by ./hack/gitlab.sh, and is overridden in CI, and
+    # a warning is issued that users should not only use ./hack/gitlab.sh for
+    # configuring test cluster available locally, such as that created by
+    # hack/cluster.sh
+    export FUNC_TEST_GITLAB_PASS="${FUNC_TEST_GITLAB_PASS:-test-password-123}"
+
+    # Generate a timestamp for use setting things which require uniqueness
+    TIMESTAMP=$(date +%Y%m%d%H%M%S 2>/dev/null || date +%s 2>/dev/null || echo "$(date)")
+
+    # Initialize coverage file
+    echo "mode: atomic" > coverage.txt
+}
+
+# -------------
+# Preconditions
+# -------------
+preconditions() {
+    echo ""
+    echo "${blue}Preconditions${reset}"
+
+    # Check if binaries are installed
+    if [ ! -d "${PROJECT_ROOT}/hack/bin" ]; then
+        echo "ERROR: hack/bin directory not found!"
+        echo "Please run ./hack/install-binaries.sh first to install required tools."
+        exit 1
+    fi
+    MISSING_BINS=""
+    for bin in kubectl kind; do
```

Comment:
Are those the only ones required? jq, stern, dapr, help, kn?

```suggestion
    for bin in kubectl kind jq stern dapr helm kn; do
```